File: WWW/Lab_4/models/event.py
from beanie import Document, PydanticObjectId
from datetime import datetime
from pydantic import BaseModel

class Event(Document):
    title: str
    description: str
    date: datetime
    location: str
    category: str
    host: PydanticObjectId

    class Settings:
        name = "events"

class EventCreate(BaseModel):
    title: str
    description: str
    date: datetime
    location: str
    category: str

File: WWW/Lab_5/lib/models/event.dart
class Event {
  final String id;
  final String title;
  final String description;
  final DateTime date;
  final String location;
  final String category;
  final String host;

  Event({
    required this.id,
    required this.title,
    required this.description,
    required this.date,
    required this.location,
    required this.category,
    required this.host,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      date: DateTime.parse(json['date']),
      location: json['location'],
      category: json['category'],
      host: json['host'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'date': date.toIso8601String(),
      'location': location,
      'category': category,
    };
  }
}
